src/meta/src/hummock/manager/mod.rs

```diff
@@ -1291,7 +1300,17 @@ where
                 branched_ssts.commit_memory();
                 current_version.apply_version_delta(&version_delta);
 
-                trigger_version_stat(&self.metrics, current_version, &versioning.version_stats);
+                let mv_id_to_all_table_ids = self
+                    .fragment_manager
+                    .get_mv_id_to_internal_table_ids_mapping()
```

Comment:
Originally I thought this approach is straightforward but after checking the fragment manager implementation, I am a little bit concerned. This function is called on `commit_epoch` and `report_compact_task`. In its implementation, it obtains the read lock of `FragmentManagerCore`. The write lock of `FragmentManagerCore` are held across etcd I/O in `fragment.rs`, which may not be release in time. Moreover, we introduce dependencies and races between ckpt/compaction and fragment manger operations (DDL, scale, ...) after this PR.

Reply:
What about moving it from `commit_epoch` and `report_compact_task` to `hummock_timer_task`? 
And optionally we may further use `try_read` in fragment.rs.